File: app/src/main/java/com/example/navigationpoc/java/ui/detail/DetailFragment.java
package com.example.navigationpoc.java.ui.detail;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.navigationpoc.R;

class DetailFragment extends Fragment {

    private TextView textView;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_detail, container, false);
        textView = root.findViewById(R.id.text_detail);
        getBundleData();
        return root;

    }


    private void getBundleData() {
        Bundle bundle = getArguments();
        if (bundle != null) {
            String mSelectedTaskOption = bundle.getString("selected_option");
            textView.setText(mSelectedTaskOption);
        }
    }
}

File: app/src/main/java/com/example/navigationpoc/kotlin/ui/detail/DetailFragment.kt
package com.example.navigationpoc.kotlin.ui.detail

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.example.navigationpoc.kotlin.MainViewModel
import com.example.navigationpoc.R

class DetailFragment : Fragment() {

    private var mSelectedTaskOption: String? = null
    private lateinit var mainViewModel: MainViewModel
    private var textView: TextView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mainViewModel =
            ViewModelProviders.of(activity!!).get(MainViewModel::class.java)
        val root = inflater.inflate(R.layout.fragment_detail, container, false)
        textView = root.findViewById(R.id.text_detail)
        getBundleData()
        return root
    }

    private fun getBundleData() {
        val bundle = arguments
        if (bundle != null) {
            mSelectedTaskOption = bundle.getString("selected_option")
            textView?.text = mSelectedTaskOption;
        }
    }
}